src/lib/adamantServerApi.js

```diff
@@ -519,6 +520,216 @@ function install (Vue) {
       this.$store.commit('ajax_end_with_error')
     })
   }
+
+  Vue.prototype.getDelegates = function (limit = constants.Delegates.ACTIVE_DELEGATES, offset = 0, votes = []) {
+    this.$store.commit('ajax_start')
+    this.$http.get(this.getAddressString() + `/api/delegates?limit=${limit}&offset=${offset}`).then(response => {
+      if (response.body.success) {
+        for (let i in response.body.delegates) {
+          let delegate = response.body.delegates[i]
+          let voted = votes.includes(delegate.address)
+          delegate._voted = voted
+          delegate.voted = voted
+          delegate.upvoted = false
+          delegate.downvoted = false
+          delegate.showDetails = false
+          delegate.forged = 0
+          delegate.status = 5
+          this.$store.commit('delegate_info', delegate)
+        }
+        this.$store.commit('ajax_end')
+      } else {
+        this.$store.commit('ajax_end_with_error')
+      }
+    }, () => {
+      this.$store.commit('ajax_end_with_error')
+    })
+  }
+  Vue.prototype.getDelegatesWithVotes = function () {
+    this.$store.commit('ajax_start')
+    this.$http.get(this.getAddressString() + '/api/accounts/delegates?address=' + this.$store.state.address).then(response => {
+      if (response.body.success) {
+        const votes = response.body.delegates.map(vote => vote.address)
+        this.$http.get(this.getAddressString() + '/api/delegates/count').then(response => {
+          if (response.body.success) {
+            this.$store.commit('ajax_end')
+            for (let i = 0; i < response.body.count % constants.Delegates.ACTIVE_DELEGATES; i++) {
+              this.getDelegates(constants.Delegates.ACTIVE_DELEGATES, i * constants.Delegates.ACTIVE_DELEGATES, votes)
+            }
+          } else {
+            this.$store.commit('ajax_end_with_error')
+          }
+        })
+      } else {
+        this.$store.commit('ajax_end_with_error')
+      }
+    }, () => {
+      this.$store.commit('ajax_end_with_error')
+    })
+  }
+  Vue.prototype.checkUnconfirmedTransactions = function () {
+    this.$store.commit('ajax_start')
+    this.$http.get(this.getAddressString() + '/api/transactions/unconfirmed').then(response => {
+      if (response.body.success) {
+        if (response.body.count === 0) {
+          this.$store.commit('set_last_transaction_status', true)
+          // this.$store.commit('ajax_end')
+        } else {
+          this.checkUnconfirmedTransactions()
+        }
+      } else {
+        this.$store.commit('ajax_end_with_error')
+      }
+    }, () => {
+      this.$store.commit('ajax_end_with_error')
+    })
+  }
+  Vue.prototype.voteForDelegates = function (votes) {
+    let keys = this.getKeypair()
+    let transaction = {
+      type: constants.Transactions.VOTE,
+      asset: {votes: votes},
+      timestamp: adamant.epochTime(),
```

Comment:
Client local time may not be in sync with the Adamant node time, which may cause the transaction to fail. `adamant-api.js` has a workaround for this.